features/admin.server-configurations.v1/forms/username-recovery-form.tsx

```diff
@@ -100,34 +106,102 @@ export const UsernameRecoveryConfigurationForm: FunctionComponent<UsernameRecove
+            "Recovery.Notification.Username.Enable":
+                (values.enableEmailBasedRecovery || values.enableSMSBasedRecovery) !== undefined
+                    ? values.enableEmailBasedRecovery || values.enableSMSBasedRecovery
+                    : (initialConnectorValues?.enableEmailBasedRecovery
+                        || initialConnectorValues?.enableSMSBasedRecovery),
+            "Recovery.Notification.Username.SMS.Enable": values.enableSMSBasedRecovery !== undefined
+                ? values.enableSMSBasedRecovery
+                : initialConnectorValues?.enableSMSBasedRecovery
         };
 
         return data;
     };
 
+    if (!initialConnectorValues){
+        return null;
+    }
+
     return (
         <div className={ "connector-form" }>
             <Form
                 id={ FORM_ID }
-                onSubmit={ null }
+                onSubmit={ (values: Record<string, any>) => onSubmit(getUpdatedConfigurations(values)) }
                 uncontrolledForm={ false }
+                initialValues={ initialConnectorValues }
             >
+                <Heading as="h4" className="mb-5">
+                    { t("extensions:manage.serverConfigurations.accountRecovery." +
+                            "passwordRecovery.recoveryOptionHeading") as ReactNode }
+                </Heading>
+
+                <Field.Checkbox
+                    className="mb-5"
+                    ariaLabel="enableEmailBasedRecovery"
+                    name="enableEmailBasedRecovery"
+                    label={ t("extensions:manage.serverConfigurations.accountRecovery." +
+                        "usernameRecovery.form.fields.enableEmailBasedRecovery.label") }
+                    required={ false }
+                    readOnly={ readOnly }
+                    width={ 10 }
+                    data-testid={ `${ testId }-email-based-recovery` }
+                    data-componentid={ `${ testId }-email-based-recovery` }
+                />
+
+                {
+                    <Message info>
+                        <Icon name="info circle" />
+                        <Trans
+                            i18nKey={
+                                "extensions:manage.serverConfigurations.accountRecovery." +
+                                "passwordRecovery.form.smsProviderWarning"
+                            }
+                        >
+                            Ensure that an
+                            <Link
+                                external={ false }
+                                onClick={ () => {
+                                    history.push(
+                                        AppConstants.getPaths().get("SMS_PROVIDER")
+                                    );
+                                } }
+                            >SMS Provider
+                            </Link>
+                            &nbsp;is configured for the OTP feature to work properly.
```

Comment:
Just make sure this &nbsp; char is not rendered in the UI. Ideally it shouldn't, but double check just to be sure :)

Reply:
Yes, this is not shown in the UI.